src/pal/src/exception/signal.cpp

```diff
@@ -283,6 +283,20 @@ static void sigsegv_handler(int code, siginfo_t *siginfo, void *context)
         record.ExceptionAddress = GetNativeContextPC(ucontext);
         record.NumberParameters = 2;
 
+        if (record.ExceptionCode == EXCEPTION_ACCESS_VIOLATION)
+        {
+            // Check if the failed access has hit a stack guard page. In such case, it
+            // was a stack probe that detected that there is not enough stack left.
+            void* stackLimit = CPalThread::GetStackLimit();
+            void* stackGuard = (void*)((size_t)stackLimit - getpagesize());
+            if ((siginfo->si_addr >= stackGuard) && (siginfo->si_addr < stackLimit))
+            {
+                // The exception happened in the page right below the stack limit,
+                // so it is a stack overflow
+                abort();
```

Comment:
Can we write "Process is terminated due to StackOverflowException." to stderr when this happens?

(to fix https://github.com/dotnet/coreclr/issues/1504)